packages/flutter/lib/src/material/bottom_app_bar_theme.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Defines a theme for [BottomAppBar] widgets.
+///
+/// Descendant widgets obtain the current [BottomAppBarTheme] object using
+/// `BottomAppBarTheme.of(context)`. Instances of [BottomAppBarTheme] can be
+/// customized with [BottomAppBarTheme.copyWith].
+///
+/// See also:
+///
+///  * [BottomAppBar], a material bottom app bar that can be customized using
+///    this [BottomAppBarTheme].
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class BottomAppBarTheme extends Diagnosticable {
+  /// Creates a dialog theme that can be used for [ThemeData.BottomAppBarTheme].
+  const BottomAppBarTheme({
+    this.color,
+    this.elevation,
+    this.shape,
+  });
+
+  /// Default value for [BottomAppBar.color].
+  final Color color;
+
+  /// Default value for [BottomAppBar.elevation].
+  final double elevation;
+
+  /// Default value for [Dialog.shape].
+  final NotchedShape shape;
+
+  /// Creates a copy of this object but with the given fields replaced with the
+  /// new values.
+  BottomAppBarTheme copyWith({
+    Color color,
+    double elevation,
+    NotchedShape shape,
+  }) {
+    return BottomAppBarTheme(
+      color: color ?? this.color,
+      elevation: elevation ?? this.elevation,
+      shape: shape ?? this.shape,
+    );
+  }
+
+  /// The data from the closest [BottomAppBarTheme] instance given the build context.
+  static BottomAppBarTheme of(BuildContext context) {
+    return Theme.of(context).bottomAppBarTheme;
+  }
+
+  /// Linearly interpolate between two dialog themes.
+  ///
+  /// The arguments must not be null.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static BottomAppBarTheme lerp(BottomAppBarTheme a, BottomAppBarTheme b, double t) {
+    assert(t != null);
+    return BottomAppBarTheme(
+      color: Color.lerp(a?.color, b?.color, t),
+      elevation: lerpDouble(a?.elevation, b?.elevation, t),
+      shape: t < 0.5 ? a?.shape : b?.shape,
```

Comment:
Is there possibly a more smooth lerp for a shape?

If not, maybe a comment about 0.5 for the switch, instead of  like 1.0?

Reply:
instead of something like 1.0*

Reply:
Unfortunately not, since BAB's `NotchedShape` [is it's own class](https://github.com/johnsonmh/flutter/blob/d927c9331005f81157fa39dff7b5dab415ad330b/packages/flutter/lib/src/painting/notched_shapes.dart#L13) we can't use the `ShapeBorder`'s lerps. 

As for why `0.5`, in general I think the pattern for `lerp`ing is: if you can't cleverly lerp, then switch from `a` to `b` at `t == 0.5` (halfway through the lerp). For example, here's how `SliderTheme` lerps [ThumbShape](https://github.com/johnsonmh/flutter/blob/a6e901130e08c2b8000001af1a4c13165d32321a/packages/flutter/lib/src/material/slider_theme.dart#L414).

Reply:
Ack